src/quest_manager/tests/test_views.py

```diff
@@ -159,6 +161,97 @@ def test_start(self):
         # the view should have redirect to the same submission:
         self.assertRedirects(response, sub.get_absolute_url())
 
+    @tag("do")
+    def test_student_no_quests_help_text(self):
+        """ 
+            When student has no quests but have:
+
+            quests in the In Progress Tab:
+                "You have no new quests available, but you can find some quests you have already started in your 'In Progress' tab above."
+
+            hidden quests:
+                "You have no new quests available, but you do have hidden quests which you can view by hitting the 'Show Hidden Quests' button above."
+
+            No in-progress and no hidden quests, but quests awaiting approval: 
+                "New quests will become available after your teacher approves your submission!"
+
+        """
+        url = reverse('quests:available')
+        user = self.test_student1
+
+        # login
+        success = self.client.login(username=user.username, password=self.test_password)
+        self.assertTrue(success)
+
+        # condition to see any of the 3
+        self.assertTrue(user.profile.has_current_course and not user.is_staff)
+
+        # make sure no available quests
+        Quest.objects.all().delete()
+        self.assertFalse(Quest.objects.get_available(user).exists())
+
+        # quest should be in 'in-progress' ===
+        quest = baker.make(Quest)
+        QuestSubmission.objects.create_submission(user, quest)
+
+        # conditions for msg to show
+        inprogress = QuestSubmission.objects.all_not_completed(user, blocking=True)  # should exist
+        available = Quest.objects.get_available(user, True)  # should not exist
+        self.assertTrue(inprogress.exists()) 
+        self.assertFalse(available.exists())
+
+        response = self.client.get(url)
+        self.assertContains(response, "No quests are available.")
+        flake8string = "You have no new quests available, but you can find some quests you have already started in your 'In Progress' tab above."
+        self.assertContains(response, flake8string)
+        QuestSubmission.objects.all().delete()
+        Quest.objects.all().delete()
+        
+        # # #quest should be in 'hidden' ===
+        # quest = Quest.objects.create(
```

Comment:
Remove all this commented out code, or put a big comment on top explaining why it's here and what it's waiting for before it can be used.

Reply:
Actually, maybe just delete this code and add a new issue on github for "Add tests to 'test_student_no_quests_help_text' " and describe the needed missing tests in the issue

Reply:
submitted an issue refrencing this: https://github.com/bytedeck/bytedeck/issues/1011